src/main/java/org/truffleruby/language/loader/FeatureLoader.java

```diff
@@ -50,10 +56,35 @@
     private TruffleObject sulongLoadLibraryFunction;
     private Map<String, String> nativeLibraryMap;
 
+    private NativeFunction getcwd;
+    private static final int PATH_MAX = 1024; // jnr-posix hard codes this value
+    private int cwdBufferSize = PATH_MAX;
+    private Pointer cwdBuffer;
+
     public FeatureLoader(RubyContext context) {
         this.context = context;
     }
 
+    public void initialize(NativePlatform nativePlatform, FinalizationService finalizationService) {
+        final TruffleNFIPlatform nfi = nativePlatform.getTruffleNFI();
+        final Object size_t_typedef = nativePlatform.getRubiniusConfiguration().get("rbx.platform.typedef.size_t");
+        final String size_t = nfi.toNFIType(StringOperations.getString((DynamicObject) size_t_typedef));
```

Comment:
This looks like something we would need elsewhere, so maybe make it a utility somewhere? Or couldn't NFI take care of handling `size_t` in the signature?

Reply:
NFI itself does not. But yes, we can add a helper in TruffleNFIPlatform, good idea, it's used in 2 places at the moment (the other is in ThreadManager).